src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/AnalysisDefaults.scala

```diff
@@ -0,0 +1,62 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Location, Tag}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.LanguageDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.LocationsExtractor
+import com.microsoft.partnercatalyst.fortis.spark.transforms.people.PeopleRecognizer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+
+/**
+  * Provides default analysis method implementations for a concrete [[Analyzer]].
+  *
+  * By implementing these traits, the concrete [[Analyzer]] explicitly enables the default implementations that are
+  * applicable to it. If a default is not enabled, the compiler will enforce that a custom implementation is provided.
+  *
+  */
+private[analyzer] object AnalysisDefaults {
```

Comment:
I'm sure you considered this but why cant we define these methods in a single base class as opposed to defining so many individual traits.